dashboard/views.py

```diff
@@ -575,6 +575,8 @@ def assignments(request, course_id=0):
     df['avg_score']=df['avg_score'].fillna('Not available')
 
     df3 = df[df['towards_final_grade'] > 0.0]
+    # operate on dataframe copy to prevent Pandas "SettingWithCopyWarning" warning
+    df3 = df3.copy()
```

Comment:
This looks fine to me. It might be better to improve line 577 to be

`df3 = df.loc[df['towards_final_grade'] > 0.0].copy()`

Also what is df3? I think Sam mentioned this when we were reviewing this. It might be wordier but something more like `df_positive_grades` or something might be a better name than df3. df2 looks like `df_greater_that_percent_selection`. Perhaps this could be better refactored.

Reply:
Yes, I do agree on this and it can be named little better. The `df3` is for progress bar at the top of the assignment

Reply:
on line591 which is below  
`df2 = df[df['towards_final_grade'] >= percent_selection]`
can also be written as 
df2 = df.loc[df['towards_final_grade'] > percent_selection].copy()

Reply:
Thanks for the suggestions. I have renamed the dataframe names from df2 and df3; also added .copy() to more dataframe objects as suggested.